### PR TITLE
Resumeable MODsuit activation

### LINKED ISSUE #22834: MODsuit Activation/Deactivation Progress Gets Reset Every Time The User Is Unconscious
**Issue Description**:
MODsuit Activation/Deactivation progress gets reset back to the first stage every time the user falls unconscious. For example: Being in crit while trying to take the suit off but falling unconscious right as its about to finish requiring you to deactivate all the parts you already deactivated all over again.

**What did you expect to happen**: ``**
I expected the suit to keep activating/deactivating while unconscious OR at the very least restart the process where I left off. 

**What happened instead**:
The MODsuit started from step one of the process despite visually showing that part already activated/deactivated.

**Why is this bad/What are the consequences**:
This bug gets people killed as they cannot leave their suit despite the specific parts already having deactivated, they have to deactivate AGAIN despite those parts already being deactivated and showing visually as deactivated. Dying to a bug can give someone a big frowny face :( .

**Steps to reproduce the problem**:
1. Put on a MODsuit!
2. Deploy the MODsuit
3. Activate the MODsuit
4. Watch The engineering MOD boots seal around your feet.
5. Watch The engineering MOD gauntlets tighten around your fingers and wrists.
6. Watch The engineering MOD chestplate cinches tightly against your chest.
7. Watch The engineering MOD helmet hisses closed.
8. Fall asleep of fall unconscious
9. Wake up and realize you have to do it all over again
10. Pain.

**When did the problem start happening**:
It is in the game as of October 10, 2023

**Extra information**:

### PR DESCRIPTION
**Pull request by nicetoolbox**
## What Does This PR Do
Adds code to support resuming an interrupted (de)activation of your MODsuit.

Fixes #22834

The `/obj/item/mod/control/proc/toggle_activate` method processes activating your MODsuit. There are 4 different if statements that make changes to your boots, gauntlets, chestplate, and helmet, and then a final if statement that finalizes the activation. They each include a 2 second delay by default by using `/proc/do_after`

Since `do_after` loops and breaks if `user.stat` is truthy, the user being unconscious will make `user.stat` evaluate to 1 and do_after will return false, interrupting the part sealing.

This PR adds a check that the part is not already sealed before calling do_after. If a part is already sealed then no delay occurs, and the next step happens. There is always a final do_after call that will call finish_activation, but you won't have to seal any parts again. (If you are deactivating the suit we check the part is not already deactivated)

## Why It's Good For The Game

I think it doesn't make sense that a part would have to be sealed again, after it already displays the text "X wraps around your skin", and sounds annoying

## Testing

1. Equip captain's MODsuit
2. Deploy
3. Start activation
4. Wait until 1 or 2 parts are sealed
5. Use right click variable editor to give myself 500 damage (Dies)
6. Use right click rejuvenate 
7. Start activation
8. See that the 1 or 2 parts that were already sealed were skipped
9. See that the activation process was completed normally

The deactivation process was tested similarly.

Note: I think if the skin was changed this might break, since it reads the skin name, instead of a is_sealed variable, but I don't think any code is changing the skin at the moment, or if you did change the skin I think you'd have to remove the suit first?

## Changelog
:cl:
fix: Fix activating MODsuits while falling unconscious a bunch to save activation progress
/:cl:

### CODE REVIEW

File: code/modules/mod/mod_activation.dm [hunk 1]
```diff
@@ -191,6 +191,9 @@
 	SEND_SIGNAL(src, COMSIG_MOD_TOGGLED, user)
 	return TRUE
 
+/obj/item/mod/control/proc/is_sealed(obj/item/clothing/part)
+	return part.icon_state == "[skin]-[part.base_icon_state]-sealed"
```
Comment (CinnamonSnowball) [inline code/modules/mod/mod_activation.dm:195]: This is pretty scuffed, please consider adding a variable or some other kind of control flag that keeps track of whether a part was sealed

**Comment by CinnamonSnowball** [comment]
Also, wouldn't it be easier to simply make `do_after()`'s `stat` check optional via extra argument?

**Comment by nicetoolbox** [comment]
> Also, wouldn't it be easier to simply make `do_after()`'s `stat` check optional via extra argument, or better yet, by putting it into the "default checks" it stuffs into `extra_checks`?

I think having it continue while you unconscious would be convenient. Do you think it was just an oversight that it doesn't continue that way right now?

**Comment by CinnamonSnowball** [comment]
I assume it works the way it works right now because most `do_after`s _do_ require the user to stay conscious and able to perform whatever task they're doing, I think it's just that nobody ever bothered to make it optional

File: code/modules/mod/mod_activation.dm [hunk 2]
```diff
@@ -157,26 +157,10 @@
...
-			seal_part(chestplate, seal = !active)
-	if(do_after(wearer, activation_step_time, FALSE, target = src, allow_moving = TRUE))
-		if(has_wearer())
-			to_chat(wearer, "<span class='notice'>[helmet] hisses [active ? "open" : "closed"].</span>")
-			playsound(src, 'sound/mecha/mechmove03.ogg', 25, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
-			seal_part(helmet, seal = !active)
+	seal_boots(seal = !active)
+	seal_gauntlets(seal = !active)
+	seal_chestplate(seal = !active)
+	seal_helmet(seal = !active)
```
Comment (DGamerL) [inline code/modules/mod/mod_activation.dm:163]:
```suggestion
	var/seal = !active
	seal_boots(seal)
	seal_gauntlets(seal)
	seal_chestplate(seal)
	seal_helmet(seal)
```

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]
please address conflicts and respond to requested changes

**Comment by S34NW** [comment]
Despite being asked to make changes, this PR has gone so stale it's grown mould. Feel free to re-open the PR once the requested changes have been made.

**Pull request closed without merging**